<a href="https://colab.research.google.com/github/ReneSRuiz/Mineria_de_datos/blob/main/Ej_Limpieza_11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Equipo:** 11   
**Grupo:** 001  
**Frecuencia:** Lunes  
**Integrantes:**
* Alejandro Uriel García Alanis 1886968 
* Jorge Eduardo Estrada Dávila 1741943 
* Nestor Misael Paz Reyes 1559508
* Elisa González García 1858207 
* Rene Sobrevilla Ruiz 1941452  
  
**Datos a utilizar:** Clasificación de plantas (https://www.kaggle.com/uciml/iris)


**Tarea de limpieza de datos:**

* **Importar librerias y base de datos.**  
Primero importaremos las librerias de pandas y numpy. Después crearemos nuestro dataframe con el csv de los datos de las Iris llamada "DatosIris.csv". (link de los datos https://www.kaggle.com/uciml/iris) (Se cambió de nombre al descargarse)


In [ ]:
#importando las librerias
import pandas as pd
import numpy as np
#Nuestros datos a trabajar son
df = pd.read_csv("DatosIris.csv", index_col="Id")

* **Visualizar los datos.**  
Aquí veremos los primeros 3 datos, los últimos 3 datos y al final, los nombres de las columnas.


In [ ]:
#Veamos los primeros 3 datos
print(df.head(3), "\n")
#Veamos los últimos 3 datos
print(df.tail(3),"\n")
#Veamos los nombres de las columnas
print("\n", df.columns, "\n")

    SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
Id                                                                       
1             5.1           3.5            1.4           0.2  Iris-setosa
2             4.9           3.0            1.4           0.2  Iris-setosa
3             4.7           3.2            1.3           0.2  Iris-setosa 

     SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm         Species
Id                                                                           
148            6.5           3.0            5.2           2.0  Iris-virginica
149            6.2           3.4            5.4           2.3  Iris-virginica
150            5.9           3.0            5.1           1.8  Iris-virginica 


 Index(['SepalLengthCm', 'SepalWidthCm', 'PetalLengthCm', 'PetalWidthCm',
       'Species'],
      dtype='object') 



* **Descrpcion básica de columnas (tipo, elementos, celdas vacias, etc.)**  


In [ ]:
#Veamos las dimensiones de nuestro dataframe
print("Dimensiones: ", df.ndim)
#Veamos el número de filas
print("Filas: ",df.shape[0])
#Veamos el número de columnas
print("Columnas: ",df.shape[1])
#Veamos la cantidad de datos de nuestro dataframe
print("Datos totales: ", df.size, "\n")
#Veamos el tipo de datos por columna
print(df.dtypes, "\n")
#Vemos si hay datos vacios
df.isnull().sum() #Vemos que no hay datos vacios en ninguna columna

Dimensiones:  2
Filas:  150
Columnas:  5
Datos totales:  750 

SepalLengthCm    float64
SepalWidthCm     float64
PetalLengthCm    float64
PetalWidthCm     float64
Species           object
dtype: object 



SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

Vemos que la dimensión de nuestro csv es de 2 (es lo esperado), contiene 5 columnas y 150 filas por columna, con un total de 750 datos. Vemos que todos los datos son flotantes, a excepción de la última columna, la cual son objetos. Además, no se cuentan con datos nulos, por lo que está completa la información.

* **Cambiar nombres de columnas.**  
Como las columnas tienen nombres en inglés, decidimos cambiarlas a su traducción al español para recordar sus nombres de mejor manera

In [ ]:
#Cambiaremos los nombres de inglés a español
df=df.rename(columns = {'SepalLengthCm':'LargoSepalCm','SepalWidthCm':'AnchoSepalCm','PetalLengthCm':'LargoPetaloCm','PetalWidthCm':'AnchoPetaloCm'})
df.head(5)

LargoSepalCm  AnchoSepalCm  LargoPetaloCm  AnchoPetaloCm      Species
Id                                                                       
1            5.1           3.5            1.4            0.2  Iris-setosa
2            4.9           3.0            1.4            0.2  Iris-setosa
3            4.7           3.2            1.3            0.2  Iris-setosa
4            4.6           3.1            1.5            0.2  Iris-setosa
5            5.0           3.6            1.4            0.2  Iris-setosa

* **El objetivo es una clasificación con los datos, asi que hay que acomodar los datos de manera que se nos facilite esta tarea.**  
Queremos separar los datos dependiendo de la especie de iris, por lo que primero encontraremos los valores únicos de la columna de especies y, dependiendo de la cantidad de valores únicos, será nuestra clasificación por especie de Iris que tengamos.

In [ ]:
#En este caso, vamos a clasificar los datos por tipo de Iris, entonces veamos si hay valores únicos:
Especies=df.Species.unique()
print(Especies, "\n")
#Tambien, quitaremos las columnas. Como nos enfocaremos en los pétalos, podemos eliminar el Sepal:
df_nueva = df.drop(labels={'LargoSepalCm','AnchoSepalCm'},axis = 1 )
#Ahora hagamos un df por especie encontrada
Iris_setona = df_nueva[df_nueva['Species']==Especies[0]]
Iris_versicolor = df_nueva[df_nueva['Species']==Especies[1]]
Iris_virginica = df_nueva[df_nueva['Species']==Especies[2]]
#Comprobemos si se realizaron los cambios:
print(Iris_setona.head(3), "\n")
print(Iris_versicolor.head(3), "\n")
print(Iris_virginica.head(3), "\n")

['Iris-setosa' 'Iris-versicolor' 'Iris-virginica'] 

    LargoPetaloCm  AnchoPetaloCm      Species
Id                                           
1             1.4            0.2  Iris-setosa
2             1.4            0.2  Iris-setosa
3             1.3            0.2  Iris-setosa 

    LargoPetaloCm  AnchoPetaloCm          Species
Id                                               
51            4.7            1.4  Iris-versicolor
52            4.5            1.5  Iris-versicolor
53            4.9            1.5  Iris-versicolor 

     LargoPetaloCm  AnchoPetaloCm         Species
Id                                               
101            6.0            2.5  Iris-virginica
102            5.1            1.9  Iris-virginica
103            5.9            2.1  Iris-virginica 



* **Guardar otros archivos con agrupaciones.**  
Una vez teniendo los datos ya agrupados, crearemos un csv con la información de cada especie distinda para el análisis individual futuro.

In [ ]:
#Crearemos 3 archivos distintos por la agrupación que hagamos
Iris_setona.to_csv("Iris setona.csv", index=True)
Iris_versicolor.to_csv("Iris versicolor.csv", index=True)
Iris_virginica.to_csv("Iris virginica.csv", index=True)